Add C++ concept entry for destructors (#5986)
Codecademy/docs: content/cpp/concepts/destructors/destructors.md

---
Title: 'Destructors'
Description: 'A destructor is a special method in a C++ class that is automatically called when an instance of that class is destroyed.'
Subjects:
  - 'Computer Science'
  - 'Code Foundations'
Tags:
  - 'Classes'
  - 'Objects'

>   - 'Destructors'

Reviewer: 'Destructors' is not a valid tag here, refer to the complete list of allowed tags [here](https://github.com/Codecademy/docs/blob/main/documentation/tags.md).

```suggestion
  - 'Memory'
```

  - 'OOP'
CatalogContent:
  - 'learn-c-plus-plus'
  - 'paths/computer-science'
---

A **destructor** is a special method in a C++ class that is automatically called when an instance of that class is destroyed. Destructors are used to free resources that were allocated during the object's lifetime, such as dynamic memory or file handles. Unlike constructors, destructors cannot take arguments and do not return a value.

## Syntax

> A destructor method is defined by using the tilde (~) character followed by the ClassName. It has no parameters and no return type.

Reviewer 2: ```suggestion
A destructor method is defined by using the tilde (~) character followed by the `ClassName`. It has no parameters and no return type.
```

> A destructor method is defined by using the tilde (~) character followed by the `ClassName`. It has no parameters and no return type.

Reviewer: Symbols used in code should be in backticks.

```suggestion
A destructor method is defined using the tilde (`~`) character followed by the class name. It has no parameters and no return type.
```

```pseudo
class ClassName {
  public:
    ~ClassName() {
      // Destructor code
    }
};
```

## Examples

### Basic Destructor

A **basic destructor** is defined to perform cleanup tasks, such as releasing dynamically allocated memory or closing file streams.

```cpp
#include <iostream>

class MyClass {
  public:
    // Constructor
    MyClass() {
      std::cout << "The constructor was executed!\n";
    }
    // Destructor
    ~MyClass() {
      std::cout << "The destructor was executed!\n";
    }
};

int main() {
  MyClass myObject; // Constructor is called here
  // Destructor will be called automatically when myObject goes out of scope
  return 0;
}
```

> The result of the following code:

Reviewer 2: ```suggestion
The output of the above code will be:
```
